### PR TITLE
Implement the breadcrumb with PatternFly components

### PR DESCRIPTION
**Pull request by jelly**
PatternFly has a breadcrumb component which we where not using but instead used a button which meant we could not properly support middle click and needed to apply css workarounds to style them properly.

Middle mouse click now properly works and opens the clicked path in a new tab.

Closes: #641

* [x] Change the hostname to an icon and add a tooltip.

### CODE REVIEW

File: src/files-breadcrumbs.tsx [hunk 1]
```diff
@@ -207,15 +208,69 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+const PathBreadcrumbs = ({ path }: { path: string[] }) => {
+    const hostname = useHostname();
+
+    // HACK: strip extranous slashes as our breadcrumb can't handle them
+    if (path.length >= 2 && path[0] === '' && path[1] === '') {
+        path = path.slice(1);
+    }
+
+    if (path.length > 1 && path[path.length - 1] === '') {
+        path = path.slice(path.length - 1);
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:220]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 2]
```diff
@@ -207,15 +208,69 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+    }
+
+    const fullPath = path.slice(1);
+    fullPath.unshift(hostname || "server");
+
+    function navigate(event: React.MouseEvent<HTMLElement>) {
+        const { button, ctrlKey, metaKey } = event;
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:227]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 3]
```diff
@@ -207,15 +208,69 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        // Typescript claims to be useful, but doesn't know shit about HTML events
+        const target = event.target as HTMLButtonElement;
+        const isAnchor = target.matches("a");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:230]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 4]
```diff
@@ -207,15 +208,69 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        if (!target.parentElement)
+            return;
+        const link = target.parentElement.getAttribute("data-location");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:234]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 5]
```diff
@@ -207,15 +208,69 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        // Let the browser natively handle non-primary click events
+        // or if the control or meta (Mac) keys are pressed
+        // ...this lets opening in a new tab or window work by default.
+        if (isAnchor && link && (button === 0 && !ctrlKey && !metaKey)) {
+            event.preventDefault();
+            cockpit.location.go("/", { path: encodeURIComponent(link) });
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:241]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 6]
```diff
@@ -207,15 +208,69 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        }
+    }
+
+    return (
+        <Breadcrumb onClick={(event) => navigate(event)}>
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:246]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 7]
```diff
@@ -207,15 +208,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+const PathBreadcrumbs = ({ path }: { path: string[] }) => {
+    const hostname = useHostname();
+
+    // HACK: strip extranous slashes as our breadcrumb can't handle them
+    if (path.length >= 2 && path[0] === '' && path[1] === '') {
+        path = path.slice(1);
+    }
+
+    if (path.length > 1 && path[path.length - 1] === '') {
+        path = path.slice(path.length - 1);
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:220]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 8]
```diff
@@ -207,15 +208,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+    }
+
+    const fullPath = path.slice(1);
+    fullPath.unshift(hostname || "server");
+
+    function navigate(event: React.MouseEvent<HTMLElement>) {
+        const { button, ctrlKey, metaKey } = event;
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:227]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 9]
```diff
@@ -207,15 +208,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        // Typescript claims to be useful, but doesn't know shit about HTML events
+        const target = event.target as HTMLButtonElement;
+        const isAnchor = target.matches("a");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:230]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 10]
```diff
@@ -207,15 +208,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        if (!target.parentElement)
+            return;
+        const link = target.parentElement.getAttribute("data-location");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:234]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 11]
```diff
@@ -207,15 +208,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        // Let the browser natively handle non-primary click events
+        // or if the control or meta (Mac) keys are pressed
+        // ...this lets opening in a new tab or window work by default.
+        if (isAnchor && link && (button === 0 && !ctrlKey && !metaKey)) {
+            event.preventDefault();
+            cockpit.location.go("/", { path: encodeURIComponent(link) });
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:241]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 12]
```diff
@@ -207,15 +208,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        }
+    }
+
+    return (
+        <Breadcrumb onClick={(event) => navigate(event)}>
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:246]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 7 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:220]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 8 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:227]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 9 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:230]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 10 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:234]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 11 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:241]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 12 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:246]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 13]
```diff
@@ -257,85 +314,71 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+          className="files-overview-header"
+          padding={{ default: "padding" }}
+        >
+            <BookmarkButton path={path} />
+            {!editMode &&
+            <Tooltip content={_("Edit path")} position={TooltipPosition.bottom}>
+                <Button
+                  variant="secondary"
+                  icon={<PencilAltIcon />}
+                  onClick={() => enableEditMode()}
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:329]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 14]
```diff
@@ -257,85 +314,71 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+                <PathBreadcrumbs path={path} />
+            </PageBreadcrumb>}
+            {editMode && newPath !== null &&
+                <TextInput
+                  autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                  id="new-path-input"
+                  value={newPath}
+                  onFocus={(event) => event.target.select()}
+                  onKeyDown={handleInputKey}
+                  onChange={(_event, value) => setNewPath(value)}
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:344]: These 8 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 15]
```diff
@@ -257,85 +314,71 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+                />}
+            <FlexItem align={{ default: 'alignRight' }}>
+                {editMode &&
+                <>
+                    <Button
+                      variant="plain"
+                      icon={<CheckIcon className="breadcrumb-edit-apply-icon" />}
+                      onClick={changePath}
+                      className="breadcrumb-button-edit-apply"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:353]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 16]
```diff
@@ -257,85 +314,71 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+                    />
+                    <Button
+                      variant="plain"
+                      icon={<TimesIcon />}
+                      onClick={() => cancelPathEdit()}
+                      className="breadcrumb-button-edit-cancel"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:359]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 17]
```diff
@@ -207,15 +210,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+const PathBreadcrumbs = ({ path }: { path: string[] }) => {
+    const hostname = useHostname();
+
+    // HACK: strip extranous slashes as our breadcrumb can't handle them
+    if (path.length >= 2 && path[0] === '' && path[1] === '') {
+        path = path.slice(1);
+    }
+
+    if (path.length > 1 && path[path.length - 1] === '') {
+        path = path.slice(path.length - 1);
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:222]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 18]
```diff
@@ -207,15 +210,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+    }
+
+    const fullPath = path.slice(1);
+    fullPath.unshift(hostname || "server");
+
+    function navigate(event: React.MouseEvent<HTMLElement>) {
+        const { button, ctrlKey, metaKey } = event;
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:229]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 19]
```diff
@@ -207,15 +210,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        // Typescript claims to be useful, but doesn't know shit about HTML events
+        const target = event.target as HTMLButtonElement;
+        const isAnchor = target.matches("a");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:232]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 20]
```diff
@@ -207,15 +210,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        if (!target.parentElement)
+            return;
+        const link = target.parentElement.getAttribute("data-location");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:236]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 21]
```diff
@@ -207,15 +210,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        // Let the browser natively handle non-primary click events
+        // or if the control or meta (Mac) keys are pressed
+        // ...this lets opening in a new tab or window work by default.
+        if (isAnchor && link && (button === 0 && !ctrlKey && !metaKey)) {
+            event.preventDefault();
+            cockpit.location.go("/", { path: encodeURIComponent(link) });
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:243]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 22]
```diff
@@ -207,15 +210,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        }
+    }
+
+    return (
+        <Breadcrumb onClick={(event) => navigate(event)}>
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:248]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 23]
```diff
@@ -257,85 +316,72 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+                </>}
+            </div>
+            {editMode && newPath !== null &&
+                <TextInput
+                  autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                  id="new-path-input"
+                  value={newPath}
+                  onFocus={(event) => event.target.select()}
+                  onKeyDown={handleInputKey}
+                  onChange={(_event, value) => setNewPath(value)}
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:349]: These 8 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 24]
```diff
@@ -257,85 +316,72 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+                />}
+            {editMode &&
+                <>
+                    <Button
+                      variant="plain"
+                      icon={<CheckIcon className="breadcrumb-edit-apply-icon" />}
+                      onClick={changePath}
+                      className="breadcrumb-button-edit-apply"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:357]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 25]
```diff
@@ -257,85 +316,72 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
                     />
-                </FlexItem>
-            </Flex>
-        </PageBreadcrumb>
+                    <Button
+                      variant="plain"
+                      icon={<TimesIcon />}
+                      onClick={() => cancelPathEdit()}
+                      className="breadcrumb-button-edit-cancel"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:363]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 1 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:220]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 2 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:227]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 3 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:230]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 4 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:234]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 5 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:241]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 6 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:246]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 7 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:220]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 8 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:227]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 9 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:230]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 10 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:234]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 11 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:241]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 12 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:246]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 26]
```diff
@@ -257,85 +314,72 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+                </>}
+            </div>
+            {editMode && newPath !== null &&
+                <TextInput
+                  autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                  id="new-path-input"
+                  value={newPath}
+                  onFocus={(event) => event.target.select()}
+                  onKeyDown={handleInputKey}
+                  onChange={(_event, value) => setNewPath(value)}
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:347]: These 8 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 27]
```diff
@@ -257,85 +314,72 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
+                />}
+            {editMode &&
+                <>
+                    <Button
+                      variant="plain"
+                      icon={<CheckIcon className="breadcrumb-edit-apply-icon" />}
+                      onClick={changePath}
+                      className="breadcrumb-button-edit-apply"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:355]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 28]
```diff
@@ -257,85 +314,72 @@ export function FilesBreadcrumbs({ path, showHidden, setShowHidden }: { path: st
...
                     />
-                </FlexItem>
-            </Flex>
-        </PageBreadcrumb>
+                    <Button
+                      variant="plain"
+                      icon={<TimesIcon />}
+                      onClick={() => cancelPathEdit()}
+                      className="breadcrumb-button-edit-cancel"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:361]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 29]
```diff
@@ -205,15 +205,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+const PathBreadcrumbs = ({ path }: { path: string[] }) => {
+    const hostname = useHostname();
+
+    // HACK: strip extranous slashes as our breadcrumb can't handle them
+    if (path.length >= 2 && path[0] === '' && path[1] === '') {
+        path = path.slice(1);
+    }
+
+    if (path.length > 1 && path[path.length - 1] === '') {
+        path = path.slice(path.length - 1);
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:217]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 30]
```diff
@@ -205,15 +205,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+    }
+
+    const fullPath = path.slice(1);
+    fullPath.unshift(hostname || "server");
+
+    function navigate(event: React.MouseEvent<HTMLElement>) {
+        const { button, ctrlKey, metaKey } = event;
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:224]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 31]
```diff
@@ -205,15 +205,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        // Typescript claims to be useful, but doesn't know shit about HTML events
+        const target = event.target as HTMLButtonElement;
+        const isAnchor = target.matches("a");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:227]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 32]
```diff
@@ -205,15 +205,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        if (!target.parentElement)
+            return;
+        const link = target.parentElement.getAttribute("data-location");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:231]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 33]
```diff
@@ -205,15 +205,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        // Let the browser natively handle non-primary click events
+        // or if the control or meta (Mac) keys are pressed
+        // ...this lets opening in a new tab or window work by default.
+        if (isAnchor && link && (button === 0 && !ctrlKey && !metaKey)) {
+            event.preventDefault();
+            cockpit.location.go("/", { path: encodeURIComponent(link) });
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:238]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 34]
```diff
@@ -205,15 +205,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        }
+    }
+
+    return (
+        <Breadcrumb onClick={(event) => navigate(event)}>
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:243]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 35]
```diff
@@ -248,66 +304,51 @@ export function FilesBreadcrumbs({ path }: { path: string[] }) {
...
+                    </PageBreadcrumb>
+                </>}
+            {editMode && newPath !== null &&
+                <TextInput
+                  autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                  id="new-path-input"
+                  value={newPath}
+                  onFocus={(event) => event.target.select()}
+                  onKeyDown={handleInputKey}
+                  onChange={(_event, value) => setNewPath(value)}
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:335]: These 8 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 36]
```diff
@@ -248,66 +304,51 @@ export function FilesBreadcrumbs({ path }: { path: string[] }) {
...
+                />}
+            {editMode &&
+                <>
+                    <Button
+                      variant="plain"
+                      icon={<CheckIcon className="breadcrumb-edit-apply-icon" />}
+                      onClick={changePath}
+                      className="breadcrumb-button-edit-apply"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:343]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 37]
```diff
@@ -248,66 +304,51 @@ export function FilesBreadcrumbs({ path }: { path: string[] }) {
...
+                    />
+                    <Button
+                      variant="plain"
+                      icon={<TimesIcon />}
+                      onClick={() => cancelPathEdit()}
+                      className="breadcrumb-button-edit-cancel"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:349]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 38]
```diff
@@ -205,15 +182,72 @@ function BookmarkButton({ path }: { path: string[] }) {
...
 
+const PathBreadcrumbs = ({ path }: { path: string[] }) => {
+    // HACK: strip extranous slashes as PF's breadcrumb can't handle them
+    // Refactor the path to be the fullpath not an array of strings
+    if (path.length >= 2 && path[0] === '' && path[1] === '') {
+        path = path.slice(1);
+    }
+
+    if (path.length > 1 && path[path.length - 1] === '') {
+        path = path.slice(path.length - 1);
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:193]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 39]
```diff
@@ -205,15 +182,72 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+    }
+
+    function navigate(event: React.MouseEvent<HTMLElement>) {
+        const { button, ctrlKey, metaKey } = event;
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:197]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 40]
```diff
@@ -205,15 +182,72 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        // Typescript claims to be useful, but doesn't know shit about HTML events
+        const target = event.target as HTMLButtonElement;
+        const isAnchor = target.matches("a");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:200]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 41]
```diff
@@ -205,15 +182,72 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        if (!target.parentElement)
+            return;
+        const link = target.parentElement.getAttribute("data-location");
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:204]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 42]
```diff
@@ -205,15 +182,72 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+
+        // Let the browser natively handle non-primary click events
+        // or if the control or meta (Mac) keys are pressed
+        // ...this lets opening in a new tab or window work by default.
+        if (isAnchor && link && (button === 0 && !ctrlKey && !metaKey)) {
+            event.preventDefault();
+            cockpit.location.go("/", { path: encodeURIComponent(link) });
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:211]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 43]
```diff
@@ -205,15 +182,72 @@ function BookmarkButton({ path }: { path: string[] }) {
...
+        }
+    }
+
+    return (
+        <Breadcrumb onClick={(event) => navigate(event)}>
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:216]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 44]
```diff
@@ -248,66 +282,51 @@ export function FilesBreadcrumbs({ path }: { path: string[] }) {
...
+                    </PageBreadcrumb>
+                </>}
+            {editMode && newPath !== null &&
+                <TextInput
+                  autoFocus // eslint-disable-line jsx-a11y/no-autofocus
+                  id="new-path-input"
+                  value={newPath}
+                  onFocus={(event) => event.target.select()}
+                  onKeyDown={handleInputKey}
+                  onChange={(_event, value) => setNewPath(value)}
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:313]: These 8 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 45]
```diff
@@ -248,66 +282,51 @@ export function FilesBreadcrumbs({ path }: { path: string[] }) {
...
+                />}
+            {editMode &&
+                <>
+                    <Button
+                      variant="plain"
+                      icon={<CheckIcon className="breadcrumb-edit-apply-icon" />}
+                      onClick={changePath}
+                      className="breadcrumb-button-edit-apply"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:321]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx [hunk 46]
```diff
@@ -248,66 +282,51 @@ export function FilesBreadcrumbs({ path }: { path: string[] }) {
...
+                    />
+                    <Button
+                      variant="plain"
+                      icon={<TimesIcon />}
+                      onClick={() => cancelPathEdit()}
+                      className="breadcrumb-button-edit-cancel"
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:327]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 38 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:193]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 39 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:197]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 40 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:200]: These 2 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 41 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:204]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 42 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:211]: These 3 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 43 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:216]: This added line is not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 44 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:313]: These 8 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 45 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:321]: These 5 added lines are not executed by any test.

File: src/files-breadcrumbs.tsx (hunk 46 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:327]: These 5 added lines are not executed by any test.

**Comment by jelly** [comment]
image

One more issue

**Comment by garrett** [comment]
I never saw the alignment issue, as it's not a problem on Firefox. This seems like an issue that's Chrome-specific. It's easy to fix though, as you just need to add `align-items: center;` to the `.pf-v5-c-page__main-section` element... We probably want to add it to `.files-overview-header` specifically, if it's Cockpit-specific, unless this is some weird PF bug in Chrome (probably not?).

File: src/files-breadcrumbs.tsx (hunk 38 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:193]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-breadcrumbs.tsx:193]: These 2 added lines are not executed by any test.

Comment (cockpituous) [inline src/files-breadcrumbs.tsx:193]: These 2 added lines are not executed by any test.

**Comment by jelly** [comment]
> I never saw the alignment issue, as it's not a problem on Firefox. This seems like an issue that's Chrome-specific. It's easy to fix though, as you just need to add `align-items: center;` to the `.pf-v5-c-page__main-section` element... We probably want to add it to `.files-overview-header` specifically, if it's Cockpit-specific, unless this is some weird PF bug in Chrome (probably not?).
> 
> We also have the bakground color issues for both the dropdown and the breadcrumbs. It should be the same as the main background color for both of them. I'm not sure if this is a PF specific bug or not (perhaps).

Fixed all of these except the bookmark button as that was already so in main.

**Comment by garrett** [comment]
> Fixed all of these except the bookmark button as that was already so in main.

What do you mean? Main has the background color issue too:

image

**Review by garrett — APPROVED** [review]
I think this PR's fine to merge from my point of view! :+1: 

(We can do a fix-up outside of this PR for that bookmarks button for dark mode.)

File: src/app.scss [hunk 47]
```diff
@@ -222,3 +222,15 @@
         padding-block-start: 0;
     }
 }
+
+// // FIXME: Promote the CSS bedlow to overrides, open PF issues // //
+
+// PatternFly always adds a margin after images inside of widgets with pf-m-end, which is incorrect when it's the last element
+.pf-v5-c-button__icon.pf-m-start:last-child {
+    margin-inline-end: 0;
+}
```
Comment (garrett) [inline src/app.scss:239]: PF-React issue. It always adds margin, either at the beginning or end, regardless of if there is text or not. (It's after in the default.)

Here's the issue illustrated on the PF Docs website, on the React widget.

image

File: src/files-breadcrumbs.tsx [hunk 48]
```diff
@@ -205,15 +182,71 @@ function BookmarkButton({ path }: { path: string[] }) {
...
 
+const PathBreadcrumbs = ({ path }: { path: string[] }) => {
+    // HACK: strip extraneous slashes as PF's breadcrumb can't handle them
+    // Refactor the path to be the fullpath not an array of strings
+    if (path.length >= 2 && path[0] === '' && path[1] === '') {
+        path = path.slice(1);
+    }
+
+    if (path.length > 1 && path[path.length - 1] === '') {
+        path = path.slice(path.length - 1);
```
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:193]: These 2 added lines are not executed by any test.

**Review by martinpitt — APPROVED** [review]
LGTM, thanks! Not sure what you want to do about the PF bugs -- block the landing on reporting those, or give up on that.

**Comment by martinpitt** [comment]
I spot-checked several [pixel test failures](https://cockpit-logs.us-east-1.linodeobjects.com/pull-670-eb3d44a9-20240730-085137-fedora-40/log.html), they look nice now, thanks! They still need to be mopped up.

File: src/files-breadcrumbs.tsx (hunk 48 above)
Comment (cockpituous) [inline src/files-breadcrumbs.tsx:193]: These 2 added lines are not executed by any test.

**Review by garrett — APPROVED** [review]
Approved.

We should do a few follow-ups:

1. Fix the background of the bookmarks button.
2. Convert the overrides to PF5 overrides in Cockpit (at least the first one) and file PF issues (at least for the first one, where it should be more obvious and easy).
   - The second one will likely require at least a day of my time just "yak shaving" to provide examples across all dropdown/select widgets and convince them — and it'll be ignored in deference of PF6 anyway, where it's going to already be different anyway. So I really want to shy away from that one. :unamused:
   - The first is very straightforward, however.

But I'm happy with this PR as-is. :+1:

**Comment by leomoty** [comment]
the `pf-m-current` (aka current folder) is middle button clickable but it has no text decoration, that doesnt feel consistent

**Pull request merged**